api/environments/job_handler.go

```diff
@@ -6,339 +6,423 @@ import (
-	log "github.com/sirupsen/logrus"
-	batchv1 "k8s.io/api/batch/v1"
+	radixLabels "github.com/equinor/radix-operator/pkg/apis/utils/labels"
 	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/labels"
-	"k8s.io/apimachinery/pkg/selection"
-	"k8s.io/client-go/kubernetes"
-)
-
-const (
-	k8sJobNameLabel = "job-name" // A label that k8s automatically adds to a Pod created by a Job
 )
 
 // GetJobs Get jobs
-func (eh EnvironmentHandler) GetJobs(appName, envName, jobComponentName string) ([]deploymentModels.
-	ScheduledJobSummary, error) {
-	namespace := operatorUtils.GetEnvironmentNamespace(appName, envName)
-	kubeClient := eh.kubeUtil.KubeClient()
-	jobs, err := getSingleJobs(kubeClient, namespace, jobComponentName)
+func (eh EnvironmentHandler) GetJobs(appName, envName, jobComponentName string) ([]deploymentModels.ScheduledJobSummary, error) {
+	jobs, err := eh.getJobs(appName, envName, jobComponentName)
 	if err != nil {
 		return nil, err
 	}
-	jobPodLabelSelector := labels.Set{
-		kube.RadixJobTypeLabel: kube.RadixJobTypeJobSchedule,
-	}
-	podList, err := getPodsForSelector(kubeClient, namespace, labels.SelectorFromSet(jobPodLabelSelector))
+
+	// Backward compatibility: Get list of jobs not handled by RadixBatch
+	// TODO: Remove when there are no legacy jobs left
+	jh := legacyJobHandler{accounts: eh.accounts}
+	legacyJobs, err := jh.GetJobs(appName, envName, jobComponentName)
 	if err != nil {
 		return nil, err
 	}
-	jobPodMap, err := getJobPodsMap(podList)
+	jobs = append(jobs, legacyJobs...)
+
+	sort.SliceStable(jobs, func(i, j int) bool {
+		return utils.IsBefore(&jobs[j], &jobs[i])
+	})
+
+	return jobs, nil
+}
+
+func (eh EnvironmentHandler) getJobs(appName, envName, jobComponentName string) ([]deploymentModels.ScheduledJobSummary, error) {
+	radixBatches, err := eh.getRadixBatches(appName, envName, jobComponentName, kube.RadixBatchTypeJob)
 	if err != nil {
 		return nil, err
 	}
-	jobSummaryList := eh.getScheduledJobSummaryList(jobs, jobPodMap)
-	return jobSummaryList, nil
+
+	return eh.getScheduledJobSummaryList(radixBatches, nil), nil
 }
 
 // GetJob Gets job by name
-func (eh EnvironmentHandler) GetJob(appName, envName, jobComponentName, jobName string) (*deploymentModels.
-	ScheduledJobSummary, error) {
-	namespace := operatorUtils.GetEnvironmentNamespace(appName, envName)
-	kubeClient := eh.kubeUtil.KubeClient()
-	job, err := getJob(kubeClient, namespace, jobComponentName, jobName, kube.RadixJobTypeJobSchedule)
-	if err != nil {
-		return nil, err
+func (eh EnvironmentHandler) GetJob(appName, envName, jobComponentName, jobName string) (*deploymentModels.ScheduledJobSummary, error) {
+	if jobSummary, err := eh.getJob(appName, envName, jobComponentName, jobName); err == nil {
+		return jobSummary, nil
 	}
-	jobPodLabelSelector := labels.Set{
-		k8sJobNameLabel:        jobName,
-		kube.RadixJobTypeLabel: kube.RadixJobTypeJobSchedule,
+
```

Comment:
`err` is not handled here. It can be not only `IsNotFound`

Reply:
Added TODO